src/mescal/tapis_de_v3/jobs.clj

```diff
@@ -4,38 +4,66 @@
   (:require [clj-time.format :as tf]
             [clojure.string :as string]
             [mescal.tapis-de-v3.app-listings :as app-listings]
-            [mescal.tapis-de-v3.job-params :as params]
             [mescal.tapis-de-v3.constants :as c]
-            [mescal.util :as util]))
-
-(def ^:private timestamp-formatter
-  (tf/formatter "yyyy-MM-dd-HH-mm-ss.S"))
+            [mescal.tapis-de-v3.job-params :as params]
+            [mescal.tapis-de-v3.params :as mp]
+            [mescal.util :as util]
+            [me.raynes.fs :as fs]
+            [slingshot.slingshot :refer [try+]]))
 
 (defn- add-param-prefix
-  [prefix param]
+  [prefix param-name]
   (if-not (string/blank? (str prefix))
-    (keyword (str prefix "_" (name param)))
-    param))
-
-(defn- params-for
-  ([config param-prefix app-section]
-   (params-for config param-prefix app-section identity))
-  ([config param-prefix app-section preprocessing-fn]
-   (let [get-param-val (comp preprocessing-fn config (partial add-param-prefix param-prefix))]
-     (->> (map (comp keyword :id) app-section)
-          (map (juxt identity get-param-val))
-          (into {})
-          (remove-vals nil?)))))
-
-(defn- preprocess-param-value
+    (keyword (str prefix "_" param-name))
+    (keyword param-name)))
+
+(defn- format-input-param
+  [get-param-val {:keys [name]}]
+  (let [v (get-param-val name)]
+    {:name name :sourceUrl (when-not (string/blank? v) v)}))
+
+(defn- preprocess-file-param-value
   [v]
-  (let [v (if (map? v) (:value v) v)]
-    (when-not (= v "") v)))
+  (if-not (string/blank? v)
+    (fs/base-name v)
+    v))
+
+(defn- preprocess-absolute-path-param-value
+  [path-prefix v]
+  (if-not (string/blank? v)
+    (str (fs/file path-prefix (preprocess-file-param-value v)))
+    v))
+
+(defn- format-param
+  [get-config-val runtime input-name-set output-name-set {:keys [name arg] :as param}]
+  (let [param-type (mp/get-param-type param)
+        v (get-config-val name)
+        v (cond
+            (contains? input-name-set name) (if (= runtime "DOCKER")
+                                              (preprocess-absolute-path-param-value "/TapisInput" v)
+                                              (preprocess-file-param-value v))
+            (contains? output-name-set name) (if (= runtime "DOCKER")
+                                               (preprocess-absolute-path-param-value "/TapisOutput" v)
+                                               (preprocess-file-param-value v))
```

Comment:
Tapis apps can now run Docker tools, but Tapis will mount inputs inside the container at `/TapisInput`, and outputs in `/TapisOutput`, by default.